supabase/migrations/20240530391033_add_reports_view.sql

```diff
@@ -0,0 +1,38 @@
+CREATE OR REPLACE VIEW reports AS
+WITH first_completed_parcel AS (
+    SELECT MIN(completed_timestamp) AS start_date
+    FROM completed_parcels
+),
+list_of_weeks AS (
+ SELECT generate_series(
+        0,
+        52 * (EXTRACT(YEAR FROM CURRENT_DATE) - EXTRACT(YEAR FROM start_date)) + EXTRACT(WEEK FROM CURRENT_DATE) - EXTRACT(WEEK FROM start_date))
+     AS number_of_weeks_ago
+ FROM first_completed_parcel
+)
+SELECT
+    TO_CHAR(date_trunc('week', CURRENT_DATE) - (number_of_weeks_ago * INTERVAL '1 week'), 'YYYY-MM-DD') as week_commencing,
```

Comment:
May be worth adding a comment on what this is doing

Reply:
Done - added a few comments to explain the slightly gnarly sql here.